invoices/invoice_expiry_watcher.go

```diff
@@ -0,0 +1,215 @@
+package invoices
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/queue"
+	"github.com/lightningnetwork/lnd/zpay32"
+)
+
+// invoiceExpiry holds and invoice's payment hash and its expiry. This
+// is used to order invoices by their expiry for cancellation.
+type invoiceExpiry struct {
+	PaymentHash lntypes.Hash
+	Expiry      time.Time
+}
+
+// Less implements PriorityQueueItem.Less such that the top item in the
+// priorty queue will be the one with that expires next.
+func (e invoiceExpiry) Less(other queue.PriorityQueueItem) bool {
+	return e.Expiry.Before(other.(invoiceExpiry).Expiry)
+}
+
+// ExpiryWatcherClock is used to provide a time functions for ExpiryWatcher,
+// which can be stubbed out in tests.
+type ExpiryWatcherClock interface {
```

Comment:
another thought, since the `clock` package is primarily for testing we can expose a `MockClock` that way it doesn't have to be redefined everywhere it is used